array-reduce/main.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-console, no-unused-vars */
+
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+
+console.log('Sum of all array values:', numbers.reduce((accumulator, val) => accumulator + val, 0))
+console.log('Product of all array values:', numbers.reduce((accumulator, val) => accumulator * val, 1))
+
+
+let total = 0
+const balance = account.reduce((acc, arr) => {
+  arr.type === 'deposit' ? total += arr.amount : total -= arr.amount
+},0);
```

Comment:
Your `.reduce()` callback should always return a value to be used as the accumulator in the next iteration. Otherwise, doing this twice in a row will give different totals.